DSharpPlus/DiscordClient.cs

```diff
@@ -1647,7 +1647,7 @@ internal async Task OnMessageCreateEventAsync(DiscordMessage message, TransportU
             {
                 if (guild != null)
                 {
-                    mentionedUsers = Utilities.GetUserMentions(message).Select(xid => guild._members.TryGetValue(xid, out var member) ? member : null).Cast<DiscordUser>().ToList();
+                    mentionedUsers = Utilities.GetUserMentions(message).Select(xid => guild._members.TryGetValue(xid, out var member) ? member : new DiscordUser { Id = xid }).Cast<DiscordUser>().ToList();
```

Comment:
This creates an object without client association

Reply:
Also you forgot the second case below (the one that calls InternalGetCachedUser)